bokeh/client/session.py

```diff
@@ -281,6 +306,14 @@ def _document_changed(self, event):
             log.debug("Not sending notification back to server for a change it requested")
             return
 
+        if isinstance(event, PeriodicCallbackAdded):
+            self.add_periodic_callback(event.callback)
+            return
+
+        if isinstance(event, PeriodicCallbackRemoved):
+            self.remove_periodic_callback(event.callback)
+            return
+
```

Comment:
Your suggestion to move should_suppress_on_change into the event could be extended to fix this too by making it `event.should_sync_over_websocket` or something, and then for `PeriodicCallbackAdded`/`Removed` event, `should_sync_over_websocket would always be false, for other events it would depend on the current patch ... something like that. I am happy to clean this up after we merge this PR, the way you have it seems ok for this PR.